### PR TITLE
Add flag to replace instance label with agent hostname

### LINKED ISSUE #135: Integrations: replace `agent_hostname` label with `instance` label
#134 pointed out issues with the `agent_hostname` label. Integrations would work more nicely with existing Grafana Dashboards if integrations instead replaced the value of the `instance` label with the machine hostname. 

This is harder than it sounds; we don't want to do this manually with metric_relabeling_rules, because we'd either have to deal with an added `exported_instance` or force all integrations to not respect labels on behalf of the users. The proper way to implement this is to intercept the label at the WAL layer or replace in the remote_write configs.

### PR DESCRIPTION
**Pull request by rfratto**
Closes #135.

### CODE REVIEW

File: pkg/integrations/manager.go [hunk 1]
```diff
@@ -33,12 +34,17 @@ var (
...
 		UseHostnameLabel:          true,
+		ReplaceIntanceLabel:       true,
 	}
 )
 
 // Config holds the configuration for all integrations.
 type Config struct {
-	// When true, adds an agent_hostname label to all samples from integrations.
+	// When true, replaces the instance label with the agent hostname.
+	ReplaceIntanceLabel bool `yaml:"replace_instance_label"`
```
Comment (eamonryan) [inline pkg/integrations/manager.go:44]: Looks like a typo here - Intance

File: pkg/integrations/manager.go [hunk 2]
```diff
@@ -65,6 +71,30 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	return unmarshal((*plain)(c))
 }
 
+func (c *Config) ApplyDefaults() error {
+	if c.ReplaceIntanceLabel {
```
Comment (eamonryan) [inline pkg/integrations/manager.go:75]: same typo - Intance

**Comment by Dileep-Dora** [comment]
@rfratto was waiting for this actually, curious when this will be included in the release?

**Comment by rfratto** [comment]
Hey @Dileep-Dora, I'm planning on including this in a release sometime next week, probably closer to the end (Thursday/Friday).

**Comment by Dileep-Dora** [comment]
Okay, thanks for the quick update @rfratto , will keep following 👍

**Pull request merged**